lib/HelloSignResource.js

```diff
@@ -334,7 +334,11 @@ HelloSignResource.prototype = {
             // submit as form
             var form = new FormData();
             for(var i=0; i<data.files.length;i++) {
-                form.append("file[" + i + "]", fs.createReadStream(data.files[i]));
+                if (typeof data.files[i] === 'string') {
+                    form.append("file[" + i + "]", fs.createReadStream(data.files[i]));
+                } else if (Buffer.isBuffer(data.files[i])) {
+                    form.append("file[" + i + "]", data.files[i]);
+                }
```

Comment:
I'm commenting here as someone who doesn't regularly contribute to this repository, but I'm a little surprised this change is necessary; [`fs.createReadStream`](https://nodejs.org/api/fs.html#fs_fs_createreadstream_path_options) appears to accept both local file paths _and_ Buffers already, so I'd think that the existing code should work?

Reply:
Ah... that's something that is interesting about `fs.createReadStream()` - the Buffer, or String, has to be a name of a file on the local filesystem - it just can be passed in in different ways. But it's _always_ a local file that will be read.

What we wanted was to use the output of a PDF filler (`pdf-fill-form`), which will output the contents of the PDF to a Buffer, and go _directly_ into the creation of a `SignatureRequest`. Without this change, we would have had to write the Buffer's contents to a temp file on the system, and then use that temp filename to read it in with the existing code.

We just didn't want to have that intermediate step as we already had the file's contents.

Interestingly enough, the `FormData()` library _can_ handle a Buffer for the contents of the file to put into a `POST` message. But we had to add two things: The Buffer had to also have two properties: `.name` and `.contentType`. WIth those, and this change, we got exactly what we needed. 🙂

Reply:
Oh, thanks for explaining! It wasn't intuitive to me that `createReadStream` still interprets Buffer contents as a *filesystem path*, but it makes sense given the module that it lives in :smile:

I'll double-check with one of our regular Node maintainers to see if we can safely land this change; Thanks again!